compiler/src/compile.rs

```diff
@@ -1303,11 +1303,9 @@ impl<O: OutputStream> Compiler<O> {
                             self.emit(Instruction::Pop);
                         }
                         if let Some(false_label) = false_label {
-                            self.emit(Instruction::Duplicate);
```

Comment:
Please add the test case as you mentioned in the pull request comment.

Reply:
This `compile_test` is fairly complex. It implements [short circuit evaluation](https://en.wikipedia.org/wiki/Short-circuit_evaluation).

Please check carefully the logic of this function. I believe the `EvalContext::Expression` is for the case where an expression on the stack is desired.

It is probably a good idea to expand on our snippets here, so that we cover all cases.